Python/compile.c

```diff
@@ -2155,6 +2156,41 @@ compiler_class(struct compiler *c, stmt_ty s)
     return 1;
 }
 
+static int
+check_is_arg(expr_ty e)
+{
+    if (e->kind != Constant_kind) {
+        return 1;
+    }
+    PyObject *value = e->v.Constant.value;
+    return (value == Py_None
+         || value == Py_False
+         || value == Py_True
+         || value == Py_Ellipsis);
+}
+
+static int
+check_compare(struct compiler *c, expr_ty e)
+{
+    Py_ssize_t i, n;
+    int left = check_is_arg(e->v.Compare.left);
+    n = asdl_seq_LEN(e->v.Compare.ops);
+    for (i = 0; i < n; i++) {
+        cmpop_ty op = (cmpop_ty)asdl_seq_GET(e->v.Compare.ops, i);
+        int right = check_is_arg((expr_ty)asdl_seq_GET(e->v.Compare.comparators, i));
+        if (op == Is || op == IsNot) {
+            if (!right || !left) {
+                const char *msg = (op == Is)
+                        ? "\"is\" with a literal. Did you mean \"==\""
+                        : "\"is not\" with a literal. Did you mean \"!=\"";
```

Comment:
I think this warning looks good.  I'd like to, but can't come up with a good way to work in the phrase "identity check".

Reply:
Do you have suggestions for the wording of the news or What's New entry?

Reply:
How about: 

>The compiler now produces a SyntaxWarning when identity checks (`is` and `is not`) are used with string literals.  These can often work by accident in CPython, but are not guaranteed by the language spec.  The warning advises users to use equality tests (`==` and `!=`) instead.

Reply:
Thanks! But a warning will be emitted not just for string literals, but for numerical literals and tuples (empty tuple is a singleton in CPython, this is an implementation detail).

Reply:
You can say "certain types of literals (e.g. strings, ints)" rather than "string literals".